odoo/addons/base/models/res_config.py

```diff
@@ -158,29 +158,11 @@ def _valid_field_parameter(self, field, name):
     def copy(self, default=None):
         raise UserError(_("Cannot duplicate configuration!"))
 
-    def onchange_module(self, field_value, module_name):
-        module_sudo = self.env['ir.module.module']._get(module_name[7:])
-        if not int(field_value) and module_sudo.state in ('to install', 'installed', 'to upgrade'):
-            deps = module_sudo.downstream_dependencies()
-            dep_names = (deps | module_sudo).mapped('shortdesc')
-            message = '\n'.join(dep_names)
-            return {
-                'warning': {
-                    'title': _('Warning!'),
-                    'message': _('Disabling this option will also uninstall the following modules \n%s', message),
-                }
-            }
-        return {}
-
     def _register_hook(self):
         """ Add an onchange method for each module field. """
-        def make_method(name):
-            return lambda self: self.onchange_module(self[name], name)
-
         for name in self._fields:
             if name.startswith('module_'):
-                method = make_method(name)
-                self._onchange_methods[name].append(method)
+                self._onchange_methods[name].append(lambda self: {})
```

Comment:
The whole point of method `_register_hook()` was to add onchange methods.  If there's nothing useful left in the onchange methods, simply remove `_register_hook()` altogether.